### PR TITLE
SLVSCODE-350 Open new token generation page in SQ 9.7+

### PR DESCRIPTION
**Pull request by jblievremont**

### CODE REVIEW

File: src/lsp/protocol.ts [hunk 1]
```diff
@@ -258,4 +258,15 @@ export namespace GetRemoteProjectNames {
   );
 }
 
+export interface TokenGenerationParams {
+  serverPath?: string;
+  errorMessage?: string;
+}
+
+export namespace GetTokenGenerationParams {
+  export const type = new lsp.RequestType<string, TokenGenerationParams, null>(
```
Comment (sophio-japharidze-sonarsource) [inline src/lsp/protocol.ts:267]: Minor - why `TokenGenerationParams`? Isn't the request for `GetTokenGenerationUrl`

File: src/lsp/client.ts [hunk 2]
```diff
@@ -46,4 +46,8 @@ export class SonarLintExtendedLanguageClient extends LanguageClient {
   getRemoteProjectsForConnection(connectionId: string) {
     return this.sendRequest(protocol.GetRemoteProjectsForConnection.type, { connectionId });
   }
+
+  getTokenGenerationParams(serverUrl: string): Promise<TokenGenerationParams> {
```
Comment (sophio-japharidze-sonarsource) [inline src/lsp/client.ts:50]: Suggestion:
`getTokenGenerationUrl(baseServerUrl: string)....`

**Review by sophio-japharidze-sonarsource — APPROVED** [review]
LGTM with minor naming comments

File: src/lsp/protocol.ts (hunk 1 above)
Comment (jblievremont) [reply src/lsp/protocol.ts:267]: Yup good catch, this needs to be updated to the latest signature from the LS 👍🏻

**Pull request merged**